src/praisonai-agents/praisonaiagents/workflows/workflows.py

```
@@ -2385,14 +2403,28 @@ def execute_with_branch(step=step, idx=idx, opt_prev=optimized_previous):
                     results.append({"step": step_result["step"], "output": step_result["output"]})
                     outputs.append(step_result["output"])
                 except Exception as e:
-                    results.append({"step": f"parallel_{idx}", "output": f"Error: {e}"})
-                    outputs.append(f"Error: {e}")
+                    logger.error(f"Parallel branch {idx} failed: {e}")
+                    errors.append({"step": idx, "error": e})
+                    if parallel_step.on_failure == "fail_fast":
+                        # Cancel remaining futures
+                        for _, f in futures:
+                            f.cancel()
+                        raise WorkflowStepError(f"Parallel branch {idx} failed", cause=e)
```

Ruff (0.15.12)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)